src/third_party/dartino/gc_metadata.cc

```diff
@@ -34,7 +34,8 @@ void GcMetadata::set_up_singleton() {
   uword mark_bits_size = size >> MARK_BITS_SHIFT;
   // Ensure there is a little slack after the mark bits for the border case
   // where we check a one-word object at the end of a page for blackness.
-  mark_bits_size++;
+  // We need everything to stay word-aligned, so we add a full word of padding.
+  mark_bits_size += sizeof(uword);
```

Comment:
The slop here must be a whole word so that we can do word-aligned loads of things that come later in the metadata.